examples/platforms/cc2538/uart.c

```diff
@@ -54,14 +54,22 @@ static void processTransmit(void);
 static const uint8_t *sTransmitBuffer = NULL;
 static uint16_t sTransmitLength = 0;
 
+typedef struct RecvBuffer
+{
+    uint16_t Offset;
+    uint16_t Length;
+} RecvBuffer;
+
 static uint8_t sReceiveBuffer[kReceiveBufferSize];
```

Comment:
There is no guarantee processing one loop for OpenThread will fit into specific time window. Do you have any easy way to calculate/estimate the worst case time on a particular device? Assuming the max length Spinel command (1300 bytes) is written, it seems to very likely the 128 byte buffer might fill faster than it can be drained.